L1Trigger/L1TMuonOverlapPhase2/src/InputMakerPhase2.cc

```diff
@@ -0,0 +1,187 @@
+}
+
+void DtPhase2DigiToStubsConverterOmtf::addDTetaStubs(MuonStubPtrs2D& muonStubsInLayers,
+                                                     const L1MuDTChambThDigi& thetaDigi,
+                                                     unsigned int iProcessor,
+                                                     l1t::tftype procTyp) {
+  //in the Phase1 omtf the theta stubs are merged with the phi in the addDTphiDigi
+  //TODO implement if needed
+}
+
+bool DtPhase2DigiToStubsConverterOmtf::acceptDigi(const DTChamberId& dTChamberId,
+                                                  unsigned int iProcessor,
+                                                  l1t::tftype procType) {
+  return OMTFinputMaker::acceptDtDigi(config, dTChamberId, iProcessor, procType);
+}
+
+InputMakerPhase2::InputMakerPhase2(const edm::ParameterSet& edmParameterSet,
+                                   MuStubsInputTokens& muStubsInputTokens,
+                                   edm::EDGetTokenT<L1Phase2MuDTPhContainer> inputTokenDTPhPhase2,
+                                   const OMTFConfiguration* config,
+                                   std::unique_ptr<OmtfAngleConverter> angleConverter)
+    : OMTFinputMaker(edmParameterSet, muStubsInputTokens, config, std::move(angleConverter)) {
+  edm::LogImportant("OMTFReconstruction") << "constructing InputMakerPhase2" << std::endl;
+
+  /*  if(!edmParameterSet.getParameter<bool>("dropDTPrimitives"))
+    digiToStubsConverters.emplace_back(std::make_unique<DtDigiToStubsConverterOmtf>(config, &angleConverter, muStubsInputTokens.inputTokenDtPh, muStubsInputTokens.inputTokenDtTh));
+
+  if(!edmParameterSet.getParameter<bool>("dropCSCPrimitives"))
+    digiToStubsConverters.emplace_back(std::make_unique<CscDigiToStubsConverterOmtf>(config, &angleConverter, muStubsInputTokens.inputTokenCSC));
+
+  if(!edmParameterSet.getParameter<bool>("dropRPCPrimitives"))
+    digiToStubsConverters.emplace_back(std::make_unique<RpcDigiToStubsConverterOmtf>(config, &angleConverter, &rpcClusterization, muStubsInputTokens.inputTokenRPC));*/
+
+  if (edmParameterSet.exists("usePhase2DTPrimitives") && edmParameterSet.getParameter<bool>("usePhase2DTPrimitives")) {
+    if (edmParameterSet.getParameter<bool>("dropDTPrimitives") != true)
+      throw cms::Exception(
+          "L1TMuonOverlapPhase2 InputMakerPhase2::InputMakerPhase2 usePhase2DTPrimitives is true, but dropDTPrimitives "
+          "is not true");
+    //if the Phase2DTPrimitives are used, then the phase1 DT primitives should be dropped
+    edm::LogImportant("OMTFReconstruction") << " using Phase2 DT trigger primitives" << std::endl;
+    digiToStubsConverters.emplace_back(std::make_unique<DtPhase2DigiToStubsConverterOmtf>(
+        config, this->angleConverter.get(), inputTokenDTPhPhase2, muStubsInputTokens.inputTokenDtTh));
+  }
+}
+
+InputMakerPhase2::~InputMakerPhase2() {
+  // TODO Auto-generated destructor stub
+}
```

Comment:
Is this destructor meant to be empty or no? If it is, could the destructor just be set to default in the specification?